src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -29,17 +29,76 @@
 
 #include <drivers/hwinfo.h>
 
+#ifdef CONFIG_MCUBOOT_IMG_MANAGER
+#include <dfu/mcuboot.h>
+#endif
+
 #include <malloc.h>
 
 namespace chip {
 namespace DeviceLayer {
 
+namespace {
+
+uint8_t DetermineBootReason()
```

Comment:
Why not returning `BootReasonType`? And defining mBootReason as `BootReasonType` as well and only do a cast when printing?

Reply:
Ahh ok, `GetBootReason` accepts uint8_t.

Reply:
I don't remember precisely what was the problem, but I faced it some time ago. `BootReasonType` is defined in `DiagnosticDataProvider.h` and `GetBootReason` API is defined in the `ConfigurationManager.h`, where `DiagnosticDataProvider.h` cannot be included, so type is undefined. That's why uint8_t was used.